fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -698,6 +700,27 @@ private void checkPartitionColumnWithBaseHMSTable(SlotRef slotRef, HiveMetaStore
             }
         }
 
+        // if mv is partitioned, mv will be refreshed by partition.
+        // if mv has window functions, it should also be partitioned by and the partition by columns
+        // should contain the partition column of mv
+        private void checkWindowFunctions(
+                CreateMaterializedViewStatement statement,
+                Map<Column, Expr> columnExprMap,
+                Scope queryScope, ConnectContext context) {
+            AnalyzeState partitionExprState = new AnalyzeState();
+            ExpressionAnalyzer.analyzeExpression(statement.getPartitionRefTableExpr(), partitionExprState, queryScope, context);
+            for (Expr columnExpr : columnExprMap.values()) {
+                if (columnExpr instanceof AnalyticExpr) {
+                    AnalyticExpr analyticExpr = columnExpr.cast();
+                    if (analyticExpr.getPartitionExprs() == null
+                            || !analyticExpr.getPartitionExprs().contains(statement.getPartitionRefTableExpr())) {
+                        throw new SemanticException("window function: %s should be partitioned by partition column: %s",
```

Comment:
How about `window function %s ’s partition expressions should contain the partition column of materialized view:`